### PR TITLE
refactor: embed event bridge code into the sdk, remove mockEvents()

### PR DESCRIPTION
**Pull request by helios1138**

### CODE REVIEW

File: src/internal/setupEventBridge.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// This code serves as a bridge to allow cross-domain communication between the
+// website and the game loaded in an iframe using the postMessage api
+export function setupEventBridge() {
+  const windowRef: any = window
```
Comment (Gongreg) [inline src/internal/setupEventBridge.ts]: can we switch this to `globalThis` & also disable `window`usage in the eslint? :)

**Review by Gongreg — APPROVED** [review]

File: src/internal/setupEventBridge.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  if (windowRef.postRuneEvent) return
+
+  const challengeNumber = +(
+    new URLSearchParams(windowRef.location.search).get("challengeNumber") ?? "1"
+  )
+
+  if (!isNaN(challengeNumber)) {
+    windowRef._runeChallengeNumber = challengeNumber
+  }
```
Comment (bfelbo) [inline src/internal/setupEventBridge.ts]: Would be helpful with a comment here :)

File: src/internal/setupEventBridge.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// This code serves as a bridge to allow cross-domain communication between the
+// website and the game loaded in an iframe using the postMessage api
+export function setupEventBridge() {
+  const windowRef: any = window
+
+  if (windowRef.postRuneEvent) return
```
Comment (bfelbo) [inline src/internal/setupEventBridge.ts]: Would be helpful with a comment here :)

File: src/internal/setupEventBridge.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  windowRef.postRuneEvent = (event: any) => {
+    window.parent.postMessage({ runeGameEvent: event }, "*")
+  }
+
+  window.addEventListener("message", (msg) => {
+    if (windowRef.Rune && isRuneGameMessage(msg)) {
+      windowRef.Rune[msg.data.runeGameCommand.type]()
+    }
+  })
```
Comment (bfelbo) [inline src/internal/setupEventBridge.ts]: Would be helpful with a comment here :)

File: src/internal/setupEventBridge.ts (hunk 1 above)
Comment (helios1138) [reply src/internal/setupEventBridge.ts]: done

**Comment by helios1138** [comment]
removed mockEvents()

File: src/internal/setupEventBridge.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+// This code serves as a bridge to allow cross-domain communication between the
+// website and the game loaded in an iframe using the postMessage api
+export function setupEventBridge() {
+  if (globalThis.postRuneEvent) return
+
+  const challengeNumber = +(
+    new URLSearchParams(globalThis.location.search).get("challengeNumber") ?? "1"
+  )
+
+  if (!isNaN(challengeNumber)) {
```
Comment (mieszko4) [inline src/internal/setupEventBridge.ts]: FFTI: you can use `Number.isInteger` to be more strict about passed param.

Comment (helios1138) [reply src/internal/setupEventBridge.ts]: Thanks, didn't know about it. Changed it.

File: src/internal/setupEventBridge.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (Number.isInteger(challengeNumber)) {
+    globalThis._runeChallengeNumber = challengeNumber
+  }
+
+  globalThis.postRuneEvent = (event: any) => {
+    window.parent.postMessage({ runeGameEvent: event }, "*")
+  }
+
+  window.addEventListener("message", (msg) => {
```
Comment (Gongreg) [inline src/internal/setupEventBridge.ts]: Make sure to use globalThis in this file.

Comment (helios1138) [reply src/internal/setupEventBridge.ts]: thanks, fixed

**Pull request merged**